test/test_smilesparser.cpp

```diff
@@ -0,0 +1,61 @@
+#define BOOST_TEST_MODULE Test_smilesparser
+
+#include <boost/filesystem.hpp>
+#include <boost/test/unit_test.hpp>
+#include <unordered_set>
+
+#include "../sketcherMinimizer.h"
+#include "../sketcherMinimizerMaths.h"
+#include "../sketcherMinimizerStretchInteraction.h"
+#include "../sketcherMinimizerBendInteraction.h"
+#include "../sketcherMaeReading.h"
+#include "sketcherBasicSMILES.h"
+
+#include "maeparser/MaeConstants.hpp"
+#include "maeparser/Reader.hpp"
+
+using std::unordered_set;
+using namespace schrodinger;
+
+
+static std::unique_ptr<sketcherMinimizerMolecule> operator"" _smiles(const char * smiles, size_t len)
+{
+    return std::unique_ptr<sketcherMinimizerMolecule>(approxSmilesParse({smiles, len}));
+}
+
+BOOST_AUTO_TEST_CASE(Basics)
+{
+    auto mol = "CCCC"_smiles;
+    BOOST_TEST(mol->getAtoms().size() == 4);
+
+    mol = "CNO"_smiles;
+    auto& atoms = mol->getAtoms();
+    BOOST_TEST(atoms[0]->getAtomicNumber() == 6);
+    BOOST_TEST(atoms[1]->getAtomicNumber() == 7);
+    BOOST_TEST(atoms[2]->getAtomicNumber() == 8);
```

Comment:
nice tests! can we also please test the connectivity of two atoms using isNeighborOf in this one?

Reply:
Added